### PR TITLE
Get Timeline API docs enhancements

### LINKED ISSUE #4881: [Enhancement]: Improve Get Timeline API docs
### Description

The following API docs should be improved:

- Get Timelines or Timeline templates
- Get Timeline / Timeline template by savedObjectId
- Get Timeline template by templateTimelineId

Improvements to make:
* All three docs are currently missing **Response code** and **Example response** sections.
* **Get Timeline / Timeline template by savedObjectId** and **Get Timeline template by templateTimelineId** are missing the **Query parameters** section.
* Request examples for **Get Timeline / Timeline template by savedObjectId** and **Get Timeline template by templateTimelineId** can be updated to use mock data instead of placeholders.
* Endpoint and parameter descriptions can be improved.

### Related links / assets

Please include each of the following, if applicable:
Doc URL: 
* https://www.elastic.co/guide/en/security/current/timeline-api-get.html
* https://www.elastic.co/guide/en/security/current/_get_timeline_timeline_template_by_savedobjectid.html
* https://www.elastic.co/guide/en/security/current/_get_timeline_template_by_templatetimelineid.html

### Which documentation set needs improvement?

ESS only

### Software version

8.13

### Collaborators

_No response_

### Timeline / deliverables

8.13 release

### PR DESCRIPTION
**Pull request by natasha-moore-elastic**
Resolves #4881 by making improvements to the following API docs:
* **Get Timelines or Timeline templates**
* **Get Timeline / Timeline template by savedObjectId**
* **Get Timeline template by templateTimelineId**

### Summary of changes
All three docs:
* Improved endpoint and parameter descriptions.
* Added Response code and Example response sections.

**Get Timeline / Timeline template by savedObjectId** and **Get Timeline template by templateTimelineId** only:
* Updated request examples to use mock data instead of placeholders.
* Added the query parameter information

### Previews
* [Get Timelines or Timeline templates](https://security-docs_bk_4882.docs-preview.app.elstc.co/guide/en/security/master/timeline-api-get.html)
* [Get Timeline or Timeline template by savedObjectId](https://security-docs_bk_4882.docs-preview.app.elstc.co/guide/en/security/master/_get_timeline_or_timeline_template_by_savedobjectid.html)
* [Get Timeline template by templateTimelineId](https://security-docs_bk_4882.docs-preview.app.elstc.co/guide/en/security/master/_get_timeline_template_by_templatetimelineid.html)

### CODE REVIEW

File: docs/events/api/timeline-api-get.asciidoc [hunk 1]
```diff
@@ -1,86 +1,660 @@
...
 
 [width="100%",options="header"]
 |==============================================
 |Name |Type |Description |Required
 
-|`only_user_favorite` |Boolean a|Returns only favorite Timelines / Timeline templates if set to `true`. Default to `false`
+|`only_user_favorite` |Boolean a|If set to `true`, returns only favorite Timelines / Timeline templates. Defaults to `false`.
 |No
-|`page_index` |Number |Page number of the result
+|`page_index` |Number |The page number to return. Defaults to TBD.
```
Comment (natasha-moore-elastic) [inline docs/events/api/timeline-api-get.asciidoc:18]: @elastic/security-threat-hunting-investigations, could you provide the default values for `page_index` and ` page_size`? Thanks!

**Review by joepeeples — DISMISSED** [review]
LGTM!

File: docs/events/api/timeline-api-get.asciidoc (hunk 1 above)
Comment (michaelolo24) [reply docs/events/api/timeline-api-get.asciidoc:18]: This should be page 1 as it's requesting the full list of timelines and will return the first page by default

File: docs/events/api/timeline-api-get.asciidoc [hunk 2]
```diff
@@ -1,86 +1,660 @@
...
 |No
-|`page_size` |Number |Items per page in result
+|`page_size` |Number |The number of items to return per page. Defaults to TBD.
```
Comment (michaelolo24) [inline docs/events/api/timeline-api-get.asciidoc:20]: I believe this is 10

Comment (michaelolo24) [reply docs/events/api/timeline-api-get.asciidoc:20]: If it helps this is the default query with the values:

`api/timelines?page_size=10&page_index=1&sort_field=updated&sort_order=desc&timeline_type=default`

**Review by michaelolo24 — DISMISSED** [review]
Thanks for updating these docs!

**Comment by natasha-moore-elastic** [comment]
run docs-build

**Review by joepeeples — APPROVED** [review]

**Pull request merged**